xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -9237,4 +9237,17 @@ private XWikiAttachmentStoreInterface getXWikiAttachmentStoreInterface(String st
 
         return null;
     }
+
+    /**
+     * Compute and return the maximum authorized length for the full name (i.e. the serialized reference of the
+     * document) based on the current store limitation.
+     *
+     * @return the maximum authorized length for a document full name.
+     * @since 11.3RC1
+     */
+    @Unstable
+    public int getFullNameMaxLength()
+    {
+        return getStore().getLimitSize(this.getXWikiContext(), this.getClass(), "fullName");
```

Comment:
This actually proves the API is not in the right place and it should be in the store instead.

Reply:
There's a name in API design about this but it escapes me FTM (when you have an api that actually delegates to some other class). It's an antipattern from what I remember.

Reply:
It's also making our Model API fatter which is the opposite to what we want.